compiler/util/multiset.py

```diff
@@ -0,0 +1,52 @@
+from collections import Counter
+
+
+class Multiset:
+    """
+    In mathematics, a multiset (or bag, or mset) is a modification of the concept of a
+    set that, unlike a set, allows for multiple instances for each of its elements.
+    The number of instances given for each element is called the multiplicity of
+    that element in the multiset.
+    """
+
+    def __init__(self, iterable=None):
+        """Initialize the multiset with an optional iterable"""
+        self.counter = Counter(iterable) if iterable is not None else Counter()
```

Comment:
This also seems to work, not sure if pyright agrees
```suggestion
        self.counter = Counter(iterable)
```